src/app/server/AppDelegate.h

```diff
@@ -29,11 +29,26 @@ class AppDelegate
 public:
     virtual ~AppDelegate() {}
     /**
-     * This is called on start of session establishment process
+     * This is called when the PBKDFParamRequest is received and indicates the start of the session establishment process
      */
     virtual void OnCommissioningSessionEstablishmentStarted() {}
+
+    /**
+     * This is called when the commissioning session has been established
+     */
     virtual void OnCommissioningSessionStarted() {}
-    virtual void OnCommissioningSessionStopped(CHIP_ERROR err) {}
+
+    /**
+     * This is called when there is an error in establishing a commissioning session (such as, when an invalid passcode is provided)
+     *
+     * @param err CHIP_ERROR indicating the error that occurred during session establishment
+     */
+    virtual void OnCommissioningSessionEstablishmentError(CHIP_ERROR err) {}
+
+    /**
+     * This is called when the commissioning session establishment stops
```

Comment:
Any hints/clues about what causes it? If you got  `OnCommissioningSessionEstablishmentError`, will you get `OnCommissioningSessionStopped`?

Reply:
Yes, OnCommissioningSessionStopped is called even if OnCommissioningSessionEstablishmentError is called. I've added more docs to OnCommissioningSessionStopped to provide additional context.